blockly/src/utils/LevelUtils.java

```diff
@@ -0,0 +1,35 @@
+package utils;
+
+import core.Game;
+import core.level.Tile;
+import core.level.utils.Coordinate;
+import core.utils.Point;
+
+/** Utility class for level-related operations. */
+public class LevelUtils {
+  /**
+   * Checks if Coordinate A can see Coordinate B in the given direction.
+   *
+   * @param coordinateA The first Coordinate.
+   * @param CoordinateB The second Coordinate.
+   * @param direction The direction in which to check visibility.
+   * @return True if the two points are in line of sight, false otherwise.
+   */
+  public static boolean canSee(
+      Coordinate coordinateA, Coordinate CoordinateB, Direction direction) {
+    if (coordinateA.equals(CoordinateB)) return true;
+
+    Tile currentTile = Game.currentLevel().tileAt(coordinateA);
+    Tile targetTile = Game.currentLevel().tileAt(CoordinateB);
+    if (targetTile == null || !targetTile.canSeeThrough()) return false;
+    while (!targetTile.equals(currentTile)) {
+      if (currentTile == null || !currentTile.canSeeThrough()) return false;
```

Comment:
Hier fehlt mir eigentlich noch ein Check, ob bestimmte Entitäts-Typen im Weg sind.
z.B Steine, Monster etc....

Vermutlich braucht man dann dafür ein Component  `IchNichtDurchsichtigCompnent` oderso....
Oder ist das ein Attribut des DrawComponents? hmm das könnte verwirren

Reply:
Soll ich nicht einfach das BlockComponent nehmen?

Reply:
Monster blocken aber nicht, in die kann man ja Reinrennen. Aber to be fair, für Blockly auch egal, da müssen nur Steine blocken. 
Also ja

Reply:
Designproblem:
Beim schieben verliert ein Stein für kurze Zeit sein BlockComponent.....